dot_local/bin/executable_nix-update.tmpl

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+set -eEuo pipefail
+trap cleanup SIGINT SIGTERM ERR EXIT
+
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+
+usage() {
+  cat << EOF # remove the space between << and EOF, this is due to web plugin issue
+Usage: $(basename "${BASH_SOURCE[0]}") [-h] [-v] <command>
+
+Simple script for updating nix-env and home-manager installed packages
+
+Options:
+  -h, --help      Print this help and exit
+  -v, --verbose   Print script debug info
+  --no-color      Suppress color generation
+
+EOF
+  exit
+}
+
+cleanup() {
+  trap - SIGINT SIGTERM ERR EXIT
+  # script cleanup here
+}
+
+setup_colors() {
+  if [[ -t 2 ]] && [[ -z "${NO_COLOR-}" ]] && [[ "${TERM-}" != "dumb" ]]; then
+    NOFORMAT='\033[0m' RED='\033[0;31m' GREEN='\033[0;32m' ORANGE='\033[0;33m' BLUE='\033[0;34m' PURPLE='\033[0;35m' CYAN='\033[0;36m' YELLOW='\033[1;33m'
+  else
+    NOFORMAT='' RED='' GREEN='' ORANGE='' BLUE='' PURPLE='' CYAN='' YELLOW=''
+  fi
+}
+
+msg() {
+  echo >&2 -e "${1-}"
+}
+
+die() {
+  local msg=$1
+  local code=${2-1} # default exit status 1
+  msg "$msg"
+  exit "$code"
+}
+
+parse_params() {
+  # default values of variables set from params
+  flag=0
+  param=''
+
+  while :; do
+    case "${1-}" in
+    -h | --help) usage ;;
+    -v | --verbose) set -x ;;
+    --no-color) NO_COLOR=1 ;;
+    -?*) die "Unknown option: $1" ;;
+    *) break ;;
+    esac
+    shift
+  done
+
+  return 0
+}
+
+parse_params "$@"
+setup_colors
+
+function eval_command {
+  msg "${BLUE}$1${NOFORMAT}"
+  eval $1
+}
+
+function main {
+  HOME_MANAGER_HOME=$XDG_CONFIG_HOME/nixpkgs
+
+  eval_command "nix-channel --update"
+  eval_command "nix-env -u"
+  eval_command "nix flake update $HOME_MANAGER_HOME"
+  eval_command "home-manager switch"
+
+  FLAKE_LOCKFILE=$HOME_MANAGER_HOME/flake.lock
+  CHEZMOI_FLAKE_LOCKFILE=$(chezmoi source-path $FLAKE_LOCKFILE)
+  command cp $FLAKE_LOCKFILE $CHEZMOI_FLAKE_LOCKFILE
+  if [[ -n $(command git -C {{ .chezmoi.sourceDir }} status --short $CHEZMOI_FLAKE_LOCKFILE) ]]; then
```

Comment:
Might be useful to add a command later to `git add` and commit these changes to make life easier